environment-setup.md

```diff
@@ -0,0 +1,152 @@
+
+#### Packages
+
+* Python 3
+* [pip](https://pip.pypa.io/en/stable/installing/)
+    * Make sure you're using [the latest version of pip](https://pip.pypa.io/en/stable/installing/#upgrading-pip). (If you used homebrew to install Python, you got pip for free!)
+* [virtualenv](https://virtualenv.pypa.io/en/stable/)
+* [virtualenvwrapper](https://virtualenvwrapper.readthedocs.io/en/latest/install.html#basic-installation)
+    * virtualenvwrapper contains useful shortcut commands for all stages of virtualenv use. [Install it](https://virtualenvwrapper.readthedocs.io/en/latest/install.html#basic-installation), making sure to follow the instructions for editing your shell startup file.
+    * If you run into ownership errors and you installed Python via Homebrew, add `VIRTUALENVWRAPPER_PYTHON=/usr/local/bin/python3` to your shell startup file prior to the lines you added for virtualenvwrapper, and try again.
+
+## Docker
+
+Containers are a popular, modern approach to packaging and running software. We use the Docker engine to create, run, and destroy containers for our applications during local development. This makes it infinitely easier to manage dependencies across 5+ years of web applications.
+
+#### Packages
+
+* [Docker Community Edition](https://www.docker.com/products/container-runtime)
+     * A popular and feature-rich container engine. [Mac users, look no further for installation instructions](https://docs.docker.com/v17.09/docker-for-mac/install/). Unix users, open the Mac instructions and find your distribution in the sidebar.
+
+## Security
+
+Cryptographic security is essential for developers. You don't have to know the guts of how these tools work, but you should have them installed and get comfortable using them.
+
+To enable hard drive encryption on a Macbook, go to System Preferences > Security & Privacy > FileVault. Turn on FileVault and save a recovery key in a safe place, such as LastPass. The process will take some time, probably about a day depending on what you're doing and what model computer you have. It can run in the background as long as your computer is awake and connected to power.
+
+#### Packages
+
+* [SSH](http://linuxcommand.org/man_pages/ssh1.html)
+     * Secure Shell, a protocol for communicating securely over unsecured networks. We use it to push and pull from Git remotes and to access our servers. It comes pre-installed as a command line tool on all Mac and Linux distributions; [generate an SSH key](https://help.github.com/articles/generating-an-ssh-key/) if you haven't already and follow the instructions for adding it to your GitHub account.
+* [GPG](https://www.gnupg.org/)
+     * GNU Privacy Guard, a command line tool for encrypting and decrypting files. Mac users can `brew install gnupg2`, while enthusiasts of the "subtler arts of computing" can [compile it](https://www.gnupg.org/download/index.en.html). Then, [configure your key](https://keyring.debian.org/creating-key.html) by hand.
```

Comment:
The "subtler arts of computing" joke is cute but I think it's not very clear, and goes against our instructions about preferring package managers above:

```suggestion
     * GNU Privacy Guard, a command line tool for encrypting and decrypting files. Mac users can `brew install gnupg2`. Then, [configure your key](https://keyring.debian.org/creating-key.html) by hand.
```